python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -28,22 +28,62 @@
     "StackBrightStarsTask",
     "MeasureExtendedPsfConfig",
     "MeasureExtendedPsfTask",
+    "DetectorsInRegion",
 ]
 
 from dataclasses import dataclass
-from typing import List
 
 from lsst.afw.fits import Fits, readMetadata
 from lsst.afw.image import ImageF, MaskedImageF, MaskX
 from lsst.afw.math import StatisticsControl, statisticsStack, stringToStatisticsProperty
 from lsst.daf.base import PropertyList
 from lsst.geom import Extent2I
-from lsst.pex.config import ChoiceField, Config, ConfigurableField, DictField, Field, ListField
+from lsst.pex.config import ChoiceField, Config, ConfigurableField, Field, ListField, ConfigDictField
 from lsst.pipe.base import PipelineTaskConfig, PipelineTaskConnections, Struct, Task
 from lsst.pipe.tasks.coaddBase import subBBoxIter
 from lsst.pipe.base.connectionTypes import Input, Output
 
 
+def find_region_for_detector(det_id, detectors_focal_plane_regions):
+    """Find the focal plane region that contains a given detector.
+
+    Parameters
+    ----------
+    det_id : int
+        The detector ID.
+
+    detectors_focal_plane_regions : dict
+        A dictionary containing focal plane region names as keys, and the
+        corresponding detector IDs as the values.
+
+    Returns
+    -------
+    key: `str`
+        The name of the regio to which the given detector belongs.
+
+    Raises
+    ------
+    KeyError
+        Raised if the given detector is not included in any focal plane region.
+    """
+    for key, value_list in detectors_focal_plane_regions.items():
+        if det_id in value_list.detectors:
+            return key
+    raise KeyError(
+        "Detector %d is not included in any focal plane region.",
+        det_id,
+    )
```

Comment:
I don't know how much it matters in practice, but it might be better invert the mapping once upfront (detector ID has key and region as value, as it was before) insead of looping over a `dict` and then through a `list` for every query.

Reply:
In practice it seemingly adds very little extra runtime. But yes, I agree a cleaner way would be to construct an inverted lookup internal to these classes at runtime. We have a future refactor ticket which would be a good place to defer this effort to, so I'll make a note of that there.